### ISSUE TITLE
Bug: [prefer-optional-chain] fix leaves dangling `)`

### ISSUE THREAD

**Issue by silverwind** [typescript-eslint/typescript-eslint#9388]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&fileType=.tsx&code=IYAgZGIBTAdARuSd6wMYEog&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4XyA&tokens=false

### Repro Code

```TypeScript
a && (a.b && a.b.c)
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-optional-chain": ["error"],
  },
};
```

### tsconfig

```jsonc
{}
```

### Expected Result

```js
a?.b?.c
```

### Actual Result

```js
a?.b?.c)
```
Which is a `SyntaxError`.

**Comment by kirkwaiblinger** [comment]
Looks like you can break it simply by adding any parens as well, like so:

[Playground](https://typescript-eslint.io/play/#ts=5.4.5&fileType=.tsx&code=DYUwLgBAhgXNB2BPA3AKChAZJiAKKAdAEYCUQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4XyA&tokens=false)

```ts
let a: any;
a && (a.b)
```
fixes to 
```ts
a?.b)
```

**Comment by kirkwaiblinger** [comment]
Related: https://github.com/typescript-eslint/typescript-eslint/issues/9379
